src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -740,6 +740,39 @@ private boolean isSameParameter(CtTypeReference<?> ctParameterType, CtTypeRefere
 		return result;
 	}
 
+
+	@Override
+	public boolean hasMethod(CtMethod<?> method) {
+		if (method == null) {
+			return false;
+		}
+
+		// Checking whether the parent is the calling type.
+		try {
+			if (method.getParent() == this) {
+				return true;
+			}
+		} catch (ParentNotInitializedException ex) {
+			return false;
+		}
+
+		// Checking whether a super class has the method.
+		final CtTypeReference<?> superCl = getSuperclass();
+		if (superCl != null && superCl.getTypeDeclaration().hasMethod(method)) {
```

Comment:
getTypeDeclaration() may throw a SpoonExeception if the super class is not in the classpath. Protect with try-catch?